Cython/Utility/StringTools.c

```diff
@@ -43,10 +43,35 @@ static CYTHON_INLINE Py_ssize_t __Pyx_Py_UNICODE_ssize_strlen(const Py_UNICODE *
 
 //////////////////// InitStrings.proto ////////////////////
 
+#if CYTHON_COMPILING_IN_LIMITED_API
+static int __Pyx_InitString(__Pyx_StringTabEntry t, PyObject **str); /*proto*/
+#else
 static int __Pyx_InitStrings(__Pyx_StringTabEntry *t); /*proto*/
+#endif
 
 //////////////////// InitStrings ////////////////////
 
+#if CYTHON_COMPILING_IN_LIMITED_API
+static int __Pyx_InitString(__Pyx_StringTabEntry t, PyObject **str) {
```

Comment:
It looks like this function could be reused in the Py3 section below, so that `__Pyx_InitStrings()` would only do the looping. The Py2 code will be removed in Cython 3.1, so splitting that out already seems the best choice.

Reply:
Still TODO, will come back to this

Reply:
Done!